Update version and dependencies

mathieupotier opened the pull request:
I updated with last released dolibarr version and avoid creating PHP 5* images that are now deprecated.
I also reorder the `Dockerfile` content in order to take advantage of building cache.

tuxgasy commented:
PHP 5.6 is end of life but it's still supported by Dolibarr. I need this for some tests.

Why define DOLI_VERSION with ARG and redefine it with ENV? You will have to keep `ENV DOLI_VERSION %VERSION%`

We should only have one Dockerfile.template.

mathieupotier commented:
OK I'll create min and max version for each Dolibarr version.

Using ARG seems more clean to me, even it's a personal choice ^_^. I have no objection as you plan to create 1 Dockerfile for each Dolibarr release you wanna manage. Using ARG could help to have less Dockerfile to create : one for each PHP version instead of on for Dolibarr * PHP version.

I had created a separated Dockerfile for 7.4 because packages installation is a little bit different.
Should I add also a CONTRIBUTING.md in order to write thoses constraints for next contributors ?

- Have min and max PHP version for each Dolibarr version.
- Keep one Dolibarr template.

tuxgasy commented:
If you think a CONTRIBUTING.md file may be useful, add it.

mathieupotier commented:
Just commit CONTRIBUTING.md and your reviews changes:

1. I finally found a way to have a unique Dockerfile template even for PHP 7.4
2. I removed the ARG clause in Dockerfile to stick on replacement during `update.sh` run
3. I created a separated file `test.sh` in order to build and run a particular combination of Dolibarr and PHP, in order to ease testing.
4. I moved Dolibarr version listing into a separated file `versions.sh` which is sourced in `update.sh`, I think it could be more clear for next updates PR.

I hope all my change on `update.sh` and `test.sh` will ease your task also ... ^_^ all explained in CONTRIBUTING.md

mathieupotier commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,37 @@
+
+# How to create images
+All is done through the `update.sh` script, Dolibarr version to build are stored in `versions.sh`
+Ensure that the var `DOLIBARR_VERSIONS` in `versions.sh` is having all version you want to build.
```
```suggestion
Ensure that the var `DOLIBARR_VERSIONS` in `versions.sh` is having all versions you want to build.
```

mathieupotier commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,37 @@
+1. Create only head Dolibarr version for each Major release.
+2. Keep only one Dockerfile.template file, use `sed` value replacement in `update.sh`
+3. Keep most up to date PHP version that matches each [Dolibarr releases](https://wiki.dolibarr.org/index.php/Versions).
+4. Be careful about [supported PHP versions](https://www.php.net/supported-versions.php), try to avoid deprecated PHP Version, but only if it doesn't break rule #3.
```
```suggestion
4. Be careful about [supported PHP versions](https://www.php.net/supported-versions.php), try to avoid deprecated PHP version, but only if it doesn't break rule #3.
```
